### ISSUE TITLE
Parse form data as python dict literal when appropriate

### ISSUE THREAD

**Issue by caymard** [NickCarneiro/curlconverter#29]
Hi

First of all thanks for your useful tool.

I had an issue by parsing the following curl command:
```bash
curl -X POST -d "grant_type=client_credentials" -u "foo:bar" http://localhost/api/oauth/token/
```
Your tool returns this:
```python
data = 'grant_type=client_credentials'
requests.post('http://localhost/api/oauth/token/', data=data)
```
First, as mentioned in #24, `-u` option is not processed.
Second, the `data` field does not work in requests. To make it work, I had to change the first line for `data = {'grant_type':'client_credentials'}`.
In conclusion, your tool should have return this:
```python
data = {'grant_type': 'client_credentials'}
user = ('foo', 'bar')
requests.post('http://localhost/api/oauth/token/', data=data, auth=user)
```

Regards,

Clément

Usefull links:
* http://docs.python-requests.org/en/master/user/authentication/
* http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests

**Comment by NickCarneiro** [comment]
It looks like the only different between a string
`data = 'grant_type=client_credentials'`
and a dict literal
`data = {'grant_type': 'client_credentials'}`

is that the dict literal magically adds a content-type header.

`Content-Type: application/x-www-form-urlencoded`

I'm going to add support for dict literals when the input string looks like it's a series of url-style keys and values.

**Comment by NickCarneiro** [comment]
Fixed in 0.0.28 for python.
